v2/internal/controllers/crd_insights_autoscalesetting_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	insights "github.com/Azure/azure-service-operator/v2/api/insights/v1api20221001"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+func Test_Insights_Autoscalesetting_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVNet(tc, testcommon.AsOwner(rg), []string{"10.0.0.0/8"})
+	subnet := newSubnet(tc, vnet, "10.0.0.0/24")
+	publicIP := newPublicIp(tc, testcommon.AsOwner(rg))
+	lb := newLoadBalancerForVMSS(tc, rg, publicIP)
+
+	tc.CreateResourcesAndWait(vnet, subnet, publicIP, lb)
+
+	vmss := newVMSS20220301(tc, rg, lb, subnet)
+	tc.CreateResourceAndWait(vmss)
+
+	objectMeta := tc.MakeObjectMeta("setting")
+	setting := &insights.Autoscalesetting{
```

Comment:
Shouldn't this casing be `AutoscaleSetting`? Any idea why its not?

Reply:
I assume its because of how they case it in thier Swagger but just wanna confirm.

Reply:
Yep thats how its in their Swagger and also in the [bicep](https://learn.microsoft.com/en-us/azure/templates/microsoft.insights/autoscalesettings?pivots=deployment-language-bicep)